iOS/Flows/Conversation/Cells/ConversationMessagesCell.swift

```diff
@@ -21,21 +21,18 @@ protocol ConversationUIStateSettable {
 class ConversationMessagesCell: UICollectionViewCell, ConversationUIStateSettable, UICollectionViewDelegate {
 
     // Interaction handling
-    var handleTappedMessage: ((ConversationId, MessageId, MessageContentView) -> Void)?
-    var handleEditMessage: ((ConversationId, MessageId) -> Void)?
-    var handleCollectionViewTapped: CompletionOptional = nil 
-
-    var handleTappedConversation: ((MessageSequence) -> Void)?
-    var handleDeleteConversation: ((MessageSequence) -> Void)?
-
-    @Published var frontmostNonUserMessage: ChatMessage?
+    var messageCellDelegate: MesssageCellDelegate? {
+        get { return self.dataSource.messageCellDelegate }
+        set { self.dataSource.messageCellDelegate = newValue }
+    }
+    var handleCollectionViewTapped: CompletionOptional = nil
```

Comment:
The messages cell just passes a delegate through to its datasource. Calling the delegate functions is handled further down in the message cell itself.